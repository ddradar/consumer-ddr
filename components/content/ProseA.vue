<template>
  <NuxtLink v-if="isRelativeLink" :to="href">
    <slot />
  </NuxtLink>
  <a v-else :href="href"><slot /></a>
</template>

<script setup lang="ts">
export interface Props {
  href?: string
  blank?: boolean
}

const props = withDefaults(defineProps<Props>(), { href: '', blank: false })
const isRelativeLink = computed(() => !props.href.startsWith('http'))
</script>
