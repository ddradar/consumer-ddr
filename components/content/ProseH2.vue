<template>
  <h2 :id="id" class="title is-4">
    <slot />
    <NuxtLink :href="`#${id}`">
      <OIcon icon="link-variant" />
    </NuxtLink>
  </h2>
</template>

<script lang="ts" setup>
defineProps<{ id: string }>()
</script>
