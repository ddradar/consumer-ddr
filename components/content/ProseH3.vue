<template>
  <h3 :id="id" class="title is-5">
    <slot />
  </h3>
</template>

<script setup lang="ts">
defineProps<{ id: string }>()
</script>
