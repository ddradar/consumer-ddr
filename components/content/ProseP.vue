<template>
  <p class="block">
    <slot />
  </p>
</template>
