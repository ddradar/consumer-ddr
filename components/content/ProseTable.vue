<template>
  <table class="table is-striped">
    <slot />
  </table>
</template>
