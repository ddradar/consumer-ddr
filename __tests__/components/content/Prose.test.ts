import { mount, RouterLinkStub } from '@vue/test-utils'
import { describe, expect, test } from 'vitest'

import ProseA from '~~/components/content/ProseA.vue'
import ProseH2 from '~~/components/content/ProseH2.vue'
import ProseH3 from '~~/components/content/ProseH3.vue'
import ProseP from '~~/components/content/ProseP.vue'
import ProseTable from '~~/components/content/ProseTable.vue'

import { plugins } from '../../../__tests__/test-utils'

describe('components/content/', () => {
  const global = { plugins, stubs: { NuxtLink: RouterLinkStub } }

  describe('ProseA.vue', () => {
    test('{ href: "/songs" } renders <NuxtLink>', () => {
      const wrapper = mount(ProseA, {
        global,
        slots: { default: 'Link' },
        props: { href: '/songs' },
      })
      expect(wrapper.element).toMatchSnapshot()
    })
  })

  describe('ProseH2.vue', () => {
    test('{ id: "Header2" } renders <h2>', () => {
      const wrapper = mount(ProseH2, {
        global,
        slots: { default: 'Header2' },
        props: { id: 'Header2' },
      })
      expect(wrapper.element).toMatchSnapshot()
    })
  })

  describe('ProseH3.vue', () => {
    test('{ id: "Header3" } renders <h3>', () => {
      const wrapper = mount(ProseH3, {
        global,
        slots: { default: 'Header3' },
        props: { id: 'Header3' },
      })
      expect(wrapper.element).toMatchSnapshot()
    })
  })

  describe('ProseP.vue', () => {
    test('renders <p>', () => {
      const wrapper = mount(ProseP, { global, slots: { default: 'Content' } })
      expect(wrapper.element).toMatchSnapshot()
    })
  })

  describe('ProseTable.vue', () => {
    test('renders <table>', () => {
      const wrapper = mount(ProseTable, {
        global,
        slots: { default: 'Content' },
      })
      expect(wrapper.element).toMatchSnapshot()
    })
  })
})
