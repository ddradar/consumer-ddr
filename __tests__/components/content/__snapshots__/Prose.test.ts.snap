// Vitest Snapshot v1

exports[`components/content/ > ProseA.vue > { href: "#name" } renders <a> 1`] = `
<a
  href="#name"
>
  
  Link
  
</a>
`;

exports[`components/content/ > ProseA.vue > { href: "/songs" } renders <NuxtLink> 1`] = `
<a>
  
  Link
  
</a>
`;

exports[`components/content/ > ProseH2.vue > { id: "Header2" } renders <h2> 1`] = `
<h2
  class="title is-4"
  id="Header2"
>
  
  Header2
  
  <a
    href="#Header2"
  >
    <oicon
      icon="link-variant"
    />
  </a>
</h2>
`;

exports[`components/content/ > ProseH3.vue > { id: "Header3" } renders <h3> 1`] = `
<h3
  class="title is-5"
  id="Header3"
>
  
  Header3
  
</h3>
`;

exports[`components/content/ > ProseP.vue > renders <p> 1`] = `
<p
  class="block"
>
  
  Content
  
</p>
`;

exports[`components/content/ > ProseTable.vue > renders <table> 1`] = `
<table
  class="table"
>
  
  Content
  
</table>
`;
